src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,116 @@
+                    LocalDateTime deadlineDate = LocalDateTime.parse(deadlineString[1].substring(3), formatter);
+                    Deadline deadlineTask = new Deadline(deadlineString[0].trim(), deadlineDate);
+                    return new AddCommand(deadlineTask);
+                }
+            }
+        } else if (command.contains("event")) {
+            if (command.trim().length() == 5) {
+                throw new DukeException(ErrorMessages.EMPTY_DESCRIPTION_HEAD.getMessage() + "event" + ErrorMessages.EMPTY_DESCRIPTION_TAIL.getMessage());
+            } else {
+                String[] eventString = command.substring(6).split("/");
+                // Check if a start and end time has been provided.
+                if (eventString.length < 3) {
+                    throw new DukeException(ErrorMessages.MISSING_EVENT_TIMING.getMessage());
+                } else if (eventString.length > 3) {
+                    throw new DukeException(ErrorMessages.TOO_MANY_EVENT_TIMINGS.getMessage());
+                } else {
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(Messages.DATE_FORMAT.getMessage());
+                    LocalDateTime startDate = LocalDateTime.parse(eventString[1].substring(5).trim(), formatter);
+                    LocalDateTime endDate = LocalDateTime.parse(eventString[2].substring(3).trim(), formatter);
+                    if (endDate.isBefore(startDate)) {
+                        throw new DukeException(ErrorMessages.INVALID_END_DATE.getMessage());
+                    } else {
+                        Event eventTask = new Event(eventString[0].trim(), startDate, endDate);
+                        return new AddCommand(eventTask);
+                    }
+                }
+            }
+        } else if (command.contains("mark") && !command.contains("unmark")){
+            if (command.trim().length() == 4){
+                throw new DukeException(ErrorMessages.MISSING_TASK_NUMBER.getMessage());
+            } else {
+                int taskNumber = Integer.parseInt(command.substring(5).trim()) - 1;
+                return new MarkCommand(taskNumber);
+            }
+        } else if (command.contains("unmark")){
+            if (command.trim().length() == 6){
+                throw new DukeException(ErrorMessages.MISSING_TASK_NUMBER.getMessage());
+            } else {
+                int taskNumber = Integer.parseInt(command.substring(7).trim()) - 1;
+                return new UnmarkCommand(taskNumber);
+            }
+        } else if (command.contains("find")){
+            if (command.trim().length() == 4){
+                throw new DukeException(ErrorMessages.NO_KEYWORD_PROVIDED.getMessage());
+            } else {
+                String input = command.substring(5).trim();
+                return new FindCommand(input);
+            }
+        }
+        else {
+            throw new DukeException(ErrorMessages.INCOMPREHENSIBLE_TASK.getMessage());
```

Comment:
Thats a lot of if-else statements 😮 perhaps you could find a way to reduce the number of if-else statements to increase code readability.